### ISSUE TITLE
High-Pitched Sound During Calls

### ISSUE THREAD

**Issue by Plarpoon** [Vencord/Vesktop#317]
### Environment:
- Stable 256231 (96a39c5)
- Vencord 30bc979 (Vesktop V0.4.4)
- Electron 27.1.2
- Chromium 118.0.5993.144
- Linux 64-Bit

### Description:
During calls, other participants occasionally experience an extremely loud and high-pitched sound. This issue has persisted across the last two versions of the program.

### Reproduction Steps:
1. Join a call.
2. Engage in conversation until the high-pitched sound occurs.

### Additional Information:
A clip of the sound has been uploaded to YouTube. Note that the original sound is significantly louder, but the platform has automatically normalized the audio levels. [YouTube Link](https://youtu.be/TWJMlgfTaY4). This noise doesn't reproduce itself only once but at several moments during a regular an hour talk, it sounds like something minor but unfortunately it really hurts the ears of the people I talk with.

### Troubleshooting Attempts:
- Switched from voice activity to push-to-talk.
- Removed and re-enabled KRISP.
- Disabled automatic gain control.
- Tested Echo Cancellation option (no noticeable difference).
- Noted that this issue does not occur on the official Discord web experience or the official Discord flatpak.

### Request:
Please investigate this issue. I will provide additional information as needed.

**Comment by AAGaming00** [comment]
Have you used discord-screenaudio recently? it used to cause this issue system-wide for me until I rebooted.

**Comment by Plarpoon** [comment]
Nope, also as said this persists across the past two versions of the program. I am winging it in this case but if I remember correctly it's something that is going on for approximately the whole past month, so I have rebooted many times since then.

**Comment by AAGaming00** [comment]
Maybe try building from source and see if the electron upgrade fixed it

**Comment by Plarpoon** [comment]
Sure thing, will try during the whole of today to reproduce the bug on the source version, thanks for the idea so we can narrow further down the issue!

**Comment by Plarpoon** [comment]
Despite using nothing else but Vesktop source for the entire day, the problem persists. In fact, it has already occurred three times in just the past 15 minutes.

Interestingly, it seems to occur when I’ve been silent for a prolonged period and then suddenly start talking again. This pattern leads me to believe that the issue might be related to transitioning from a state of silence to sudden activity.

I’ve also checked the console log for any useful information, but it doesn’t seem to provide any relevant insights. Here is the [link](https://github.com/Vencord/Vesktop/files/13813986/vesktop.log) to the log for your reference.

If anyone wants me to do any specific test in real time I am up for it, even directly inside the Vencord Discord server (I am already there). Whatever can help you, I am seriously struggling to come with a reason on what this could be... Hopefully it's not another HW acceleration related bug because those are tough to resolve.

**Comment by midblep** [comment]
I've experienced the same issue. Alongside high pitched noises people have also told me they hear increasingly loud beeping or a lot of static noise. All of this goes away as soon as I use something like the official Discord web client.

**Comment by X1nto** [comment]
Are you using venmic by any chance?

**Comment by midblep** [comment]
For me it's happening even with 0 plugins or themes. Clean install of Vesktop.

**Comment by Curve** [comment]
> For me it's happening even with 0 plugins or themes. Clean install of Vesktop.

Venmic is not a plugin - It will be activated during screenshare with audio

**Comment by Plarpoon** [comment]
@X1nto Thank you for reminding me that I should have wrote down a list of active plugins, added it to the original post!

**Comment by midblep** [comment]
> > For me it's happening even with 0 plugins or themes. Clean install of Vesktop.
> 
> Venmic is not a plugin - It will be activated during screenshare with audio

Was not aware what it was. I am not screensharing or watching a screenshare when it happens. Just a normal one on one voice call.

**Comment by Curve** [comment]
> > > For me it's happening even with 0 plugins or themes. Clean install of Vesktop.
> > 
> > Venmic is not a plugin - It will be activated during screenshare with audio
> 
> Was not aware what it was. I am not screensharing or watching a screenshare when it happens. Just a normal one on one voice call.

Thanks for confirming!

**Comment by midblep** [comment]
I tried this on Windows and Vesktop on Windows does not experience the same issue. I have only been getting it on Vesktop for Linux. (Ubuntu 23.10).

**Comment by Curve** [comment]
Please let me know if this also reproduces when never starting a Screen-Share

**Comment by Curve** [comment]
> It does reproduce consistently also while having a screenshare, but not necessarily at the precise moment you start it. This I know for a fact.

This sadly doesn't really help, I'd have to know if it also reproduces when never starting a screen share in the first place

**Comment by Plarpoon** [comment]
Would you like to try in about 10 minutes in Vencord server?

**Comment by midblep** [comment]
> Please let me know if this also reproduces when never **starting** a Screen-Share

Never as in, having never done it on that PC at all? Or having never started one in that call instance?

**Comment by Curve** [comment]
> > Please let me know if this also reproduces when never **starting** a Screen-Share
> 
> Never as in, having never done it on that PC at all? Or having never started one in that call instance?

Never as in never having started a screenshare in that vesktop instance

**Comment by Curve** [comment]
> > > > Please let me know if this also reproduces when never **starting** a Screen-Share
> > > 
> > > 
> > > Never as in, having never done it on that PC at all? Or having never started one in that call instance?
> > 
> > 
> > Never as in never having started a screenshare in that vesktop instance
> 
> Yes, it happens then too. Yesterday I installed the flatpak to see if it happens there too. Clean install and I immediately go in a voice call to test, without cloud sync or settings on, and it immediately started doing weird noises.

Alright thanks for confirming this - Should not be a venmic issue then ^^

**Comment by Plarpoon** [comment]
Confirming the same thing too then!

Also quick question, is everyone who is affected by this issue running on Ubuntu (or similar)?
I have noticed @midblep saying he is running Ubuntu 23.10, I was running Kubuntu 23.10 until yesterday and now 24.04 (which is virtually identical).

If so we might finally have a clue on what it could be.

**Comment by midblep** [comment]
> Confirming the same thing too then!
> 
> Also quick question, is everyone who is affected by this issue running on Ubuntu (or similar)? I have noticed @midblep saying he is running Ubuntu 23.10, I was running Kubuntu 23.10 until yesterday and now 24.04 (which is virtually identical).
> 
> If so we might finally have a clue on what it could be.

Actually, my bad, I am also running Kubuntu, not Ubuntu. 😅
Been meaning to upgrade to 24.04 soon as well, so I suppose we have rather identical systems.

**Comment by Vendicated** [comment]
i have this issue as well, but in my case it's just an issue with my microphone that also happens outside of vesktop. on windows it's fine. either because of driver differences or because krisp works better on windows

**Comment by Plarpoon** [comment]
As said I did notice this happens primarily when I am quiet for some time and then starts talking, my guess is that it's not a KRISP issue but a gain issue.

Might be connected to an old bug I noticed, despite **automatic gain control** being disabled my microphone is still adjusted all the time up and down based on my level of voice, this bug is still happening to this day, could the two bugs be related?

**Comment by Vendicated** [comment]
the automatic gain control toggle is not implemented in vesktop. automatic gain control is always on, see https://github.com/Vencord/Vesktop/issues/161

**Comment by Plarpoon** [comment]
Yes, that's my point, could it be a bug on the way that is implemented instead of being a KRISP issue?

**Comment by RasmusRasmussen** [comment]
Same issue in Pop!_os 22.04 Ubuntu based distro, I will try restart and not start a screenshare.

**Comment by Plarpoon** [comment]
As of now it looks like 3 out of 3 people experiencing this issue have an Ubuntu based distro then.

Please if anyone using something else can write a comment I would be very happy, at least we can finally say for certain the issue is there. Try to use Vesktop on Arch, Gentoo, Fedora or whatever peeps!

**Comment by Jitzek** [comment]
> As of now it looks like 3 out of 3 people experiencing this issue have an Ubuntu based distro then.
> 
> Please if anyone using something else can write a comment I would be very happy, at least we can finally say for certain the issue is there. Try to use Vesktop on Arch, Gentoo, Fedora or whatever peeps!

I use Arch Linux x86_64 and unfortunately have the same problem:

- Stable 256231 (96a39c5) 
- Vencord D73790e (Vesktop V0.4.4)
- Electron 27.1.2
- Chromium 118.0.5993.144
- Linux 64-Bit

I'll try out the AUR package instead of the Flatpak package to see if that makes any difference.

**Comment by midblep** [comment]
Does seem to be specific to Vesktop. Not getting the same problem in the official Discord app modded with Vencord, or other ones that have it as installation options like ArmCord.

**Comment by Jitzek** [comment]
> > As of now it looks like 3 out of 3 people experiencing this issue have an Ubuntu based distro then.
> > Please if anyone using something else can write a comment I would be very happy, at least we can finally say for certain the issue is there. Try to use Vesktop on Arch, Gentoo, Fedora or whatever peeps!
> 
> I use Arch Linux x86_64 and unfortunately have the same problem:
> 
>     * Stable 256231 (96a39c5)
> 
>     * Vencord D73790e (Vesktop V0.4.4)
> 
>     * Electron 27.1.2
> 
>     * Chromium 118.0.5993.144
> 
>     * Linux 64-Bit
> 
> 
> I'll try out the AUR package instead of the Flatpak package to see if that makes any difference.

Tested it and the AUR package has the same issue.

**Comment by Ysujy** [comment]
Same issue one Arch with the vencord-desktop-git AUR package. Weirdly enough, only some participants notice the high-pitch sound.

**Comment by vynwg** [comment]
Same thing here. Sometimes only a single person hears the sound and the others don't

Vesktop info:
- Canary 260398 (9b32fcd)
- Vencord D629281 (Vesktop V0.4.4)
- Electron 28.1.1
- Chromium 120.0.6099.109

System info:
- system: `"x86_64-linux"`
- host os: `Linux 6.6.1-zen1, NixOS, 23.11 (Tapir), 23.11.2004.1e2e384c5b7c`
- multi-user?: `yes`
- sandbox: `yes`
- version: `nix-env (Nix) 2.18.1`

**Comment by midblep** [comment]
Just had my first occurrences of this exact thing happening inside Armcord with Vesktop.

**Comment by Jokeronomy** [comment]
This could have been introduced with an upgrade from the (old) vencord-desktop package, 0.4.4-3 -> 0.4.4-4 (Pacman package on Manjaro). That would seem to line up with when people in my VC's started to notice the sound (For me, I installed that on Jan 3rd, just about 3 weeks ago). I'm more than willing to do the legwork to manually figure out what commit may have introduced this bug, but I have no idea what commits were a part of that package - 0.4.4 is not even on the releases page. Given that it could realistically be a couple of hours (or more) to make the determination of it being broken or not, having that direction would really kinda be critical to determining the at-issue commit. 

Can anyone point me in the right direction?

**Comment by Curve** [comment]
> This could have been introduced with an upgrade from the (old) vencord-desktop package, 0.4.4-3 -> 0.4.4-4 (Pacman package on Manjaro). That would seem to line up with when people in my VC's started to notice the sound (For me, I installed that on Jan 3rd, just about 3 weeks ago). I'm more than willing to do the legwork to manually figure out what commit may have introduced this bug, but I have no idea what commits were a part of that package - 0.4.4 is not even on the releases page. Given that it could realistically be a couple of hours (or more) to make the determination of it being broken or not, having that direction would really kinda be critical to determining the at-issue commit. 
> 
> Can anyone point me in the right direction?

You can see the history of the aur package here:

https://aur.archlinux.org/cgit/aur.git/?h=vencord-desktop

**Comment by D3SOX** [comment]
> is there a way to force vesktop to use a specific Electron version?

Just build it locally and replace it inside package.json.

Or with an approach like the vesktop-git AUR package uses the system electron:
https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=vesktop-git

**Comment by AdivonSlav** [comment]
Can confirm I also have this issue intermittently on Vekstop v1.5.0 Flatpak on Fedora 39

**Comment by donut313** [comment]
I instantly had this issue when I first dowbloaded the app a few days ago while testing it screenshare audio actually worked. Using arch linux and wayland on kde.

**Comment by makidoll** [comment]
I downgraded electron to `25.9.4` and that seems to have solved the issue. It might be on their side.

**Comment by makidoll** [comment]
> I downgraded electron to `25.9.4` and that seems to have solved the issue. It might be on their side.

Seems like some people can still hear it but others don't. Hmm, I'm not sure what it is

**Comment by Mixaz017** [comment]
Same issue on Arch Linux / KDE / PipeWire. Vesktop is installed using `vencord-desktop-git` AUR package and all installed packages are up to date. Tweaking voice related settings in Discord didn't help.

**Comment by lewisakura** [comment]
We are aware the issue exists and it happens on a wide variety of configurations. Unfortunately more comments aren't really helping and no more productive discussion is really occuring. Please continue technical discussions on our Discord server's development channel.

Locking.

**Comment by Vendicated** [comment]
does this still happen?

**Comment by ThisNekoGuy** [comment]
I haven't personally had it happen in quite some time, but I don't know what causes it so it's hard for me to really know :/

**Comment by Jokeronomy** [comment]
I believe so, though it's perhaps rarer than it used to be. I can't really track my voice output, but I know people started commenting on me "exploding" in VC's around the time this issue was raised and that still occasionally occurs to this day.
